src/components/PriceChart.tsx

```
@@ -37,472 +47,302 @@ const PriceChart: React.FC<PriceChartProps> = ({ assetName, timeFrame = '24h', o
-        timeframeMs = 7 * 24 * 60 * 60 * 1000; // 7 days in ms
+        // Exactly 7 days ago from current time
+        targetStartTime = now.getTime() - (7 * 24 * 60 * 60 * 1000);
+        periodLabel = '7 days';
         break;
       case '30d':
-        timeframeMs = 30 * 24 * 60 * 60 * 1000; // 30 days in ms
+        // Exactly 30 days ago from current time
+        targetStartTime = now.getTime() - (30 * 24 * 60 * 60 * 1000);
+        periodLabel = '30 days';
         break;
       case 'all':
```

Biome (1.9.4)

[error] Useless case clause.

because the default clause is present:

Unsafe fix: Remove the useless case.


(lint/complexity/noUselessSwitchCase)